utilities/tools/module-indexes/module-index.tests.ps1

```diff
@@ -0,0 +1,747 @@
+                It 'Telemetry ID prefix should start with "46d3xbcp.ptn."' {
+                    $lineNumber = 2
+                    foreach ($item in $csvContent) {
+                        $item.TelemetryIdPrefix | Should -Match '^46d3xbcp.ptn.*$' -Because "TelemetryIdPrefix should start with '46d3xbcp.ptn.'. In line #$lineNumber, this is invalid. The following value have been provided: ""$($item.TelemetryIdPrefix)"""
+                    }
+                    $lineNumber++
+                }
+            } elseif ($CsvFilePath -match 'UtilityModules') {
+                It 'Telemetry ID prefix should start with "46d3xbcp.ptn."' {
+                    $lineNumber = 2
+                    foreach ($item in $csvContent) {
+                        $item.TelemetryIdPrefix | Should -Match '^46d3xbcp.utl.*$' -Because "TelemetryIdPrefix should start with '46d3xbcp.utl.'. In line #$lineNumber, this is invalid. The following value have been provided: ""$($item.TelemetryIdPrefix)"""
+                    }
+                    $lineNumber++
+                }
+            }
+
+        }
+    }
+
+    Context 'PrimaryModuleOwnerGHHandle column' {
+        It "Should have a value for 'Available' modules" {
+            $lineNumber = 2
+            foreach ($item in $csvContent) {
+                if ($item.ModuleStatus -eq 'Available :green_circle:') {
+                    $item.PrimaryModuleOwnerGHHandle | Should -Not -BeNullOrEmpty -Because "PrimaryModuleOwnerGHHandle is a required field for 'Available' modules. This should have a value in line #$lineNumber"
+                }
+                $lineNumber++
+            }
+        }
+
+        It "Should be empty for 'Orphaned' modules" {
+            $lineNumber = 2
+            foreach ($item in $csvContent) {
+                if ($item.ModuleStatus -eq 'Orphaned :eyes:') {
+                    $item.PrimaryModuleOwnerGHHandle | Should -BeNullOrEmpty -Because "PrimaryModuleOwnerGHHandle field must be empty for 'Orphaned' modules. This should not have a value in line #$lineNumber"
+                }
+                $lineNumber++
+            }
+        }
+
+        It 'Should have a value if the PrimaryModuleOwnerDisplayName also has a value' {
+            $lineNumber = 2
+            foreach ($item in $csvContent) {
+                if ($item.PrimaryModuleOwnerDisplayName -ne '') {
+                    $item.PrimaryModuleOwnerGHHandle | Should -Not -BeNullOrEmpty -Because "PrimaryModuleOwnerGHHandle should have a value if PrimaryModuleOwnerDisplayName has a value. This should have a value in line #$lineNumber"
+                }
+                $lineNumber++
+            }
+        }
+
+        It 'Should be a valid GitHub handle' {
+            $PrimaryModuleOwnerGHHandles = $csvContent.PrimaryModuleOwnerGHHandle | Where-Object { $_ -ne '' } | Select-Object -Unique
+            foreach ($item in $PrimaryModuleOwnerGHHandles) {
+                $GHUser = (& gh api users/$item) | ConvertFrom-Json
```

Comment:
Isn't this very likely to fail just as the moduleIssue assignment fails on occacion? If not (because you look for a specific user), why is it no possible to apply the same logic to the moduleIssue assignment?

Reply:
Same comment for the other, identically test for the secondary handle

Reply:
> Isn't this very likely to fail just as the moduleIssue assignment fails on occacion? If not (because you look for a specific user), why is it no possible to apply the same logic to the moduleIssue assignment?

This is a great question; I'll check with @rahalan. Btw, is there any other (easy) way to query this?

Reply:
@AlexanderSehr @matebarabas the assigning issue has nothing to do, with using the correct GH handle, which we do. The assignment process simply fails in the backend.

Reply:
I just caught up with @rahalan. He's not using the `gh api users/<username>` cmdlet in his solution, therefore has no direct data point on the success rate, but he's using a similar `gh api` invocation for grabbing GH teams and their members, and that works a 100% of the time. It appears the pagination issue is only limited to assigning issues (and PRs) to users in the Azure org. I think this comment can be resolved.